Rakefile

```diff
@@ -57,24 +57,18 @@ end
 task :build, [:build_fat] do |t, args|
   args.with_defaults(:build_fat => false)
   print_info "Building project (fat: #{args[:build_fat]})"
-  sh %Q(swift build -c release --arch x86_64 --arch arm64 --disable-sandbox --build-path #{BUILD_DIR})
+
+  # Use xcodebuild due to https://bugs.swift.org/browse/SR-15802
+  xcpretty %Q(xcodebuild -resolvePackageDependencies)
+  xcpretty %Q(xcodebuild -scheme sourcery -destination platform=macOS,arch=x86_64 -derivedDataPath #{BUILD_DIR}derivedData -archivePath #{BUILD_DIR}sourcery.xcarchive archive)
+
+  # Prepare the export direcotry
   sh %Q(rm -fr #{CLI_DIR})
   sh %Q(mkdir -p "#{CLI_DIR}bin")
-  sh %Q(mkdir -p "#{CLI_DIR}lib")
-  
-if args[:build_fat] != true  # Build thin Library
-  ARCH = `uname -m`.chomp
-  sh %Q(lipo -thin #{ARCH} lib_InternalSwiftSyntaxParser.dylib -output #{CLI_DIR}lib/lib_InternalSwiftSyntaxParser.dylib)
-else
-  sh %Q(cp lib_InternalSwiftSyntaxParser.dylib #{CLI_DIR}lib)
-end
```

Comment:
I wonder if this is still needed for `brew` version, might be not now that we have static linking

Reply:
Ah I see, so the `brew` installation works by running `rake build` (not fat) when building from the source:

https://github.com/Homebrew/homebrew-core/blob/ab5040f62247ae337489653254bd817f33645006/Formula/sourcery.rb#L20-L24

But `:fat_build` is used when running `rake build_and_deploy`, which builds the universal binary for deployment everywhere.

---

The thing is, `build_fat` was only used for stripping the unused archs from the dylib and not the cli binary itself. Before this change, we'd run `swift build --arch x86_64 --arch arm64` regardless of `build_fat` but if we actually wanted to build for the current architecture, we should just run `swift build` (or the `xcodebuild` equivalent).

Think its worth correcting this so that `brew install sourcery --build-from-source` only builds the one architecture? I can probably do that in this PR. 

I'm not too familiar with best practices for Homebrew, but I checked to see how SwiftLint is setup and it looks like it builds just the active arch so I think we can follow that (https://github.com/Homebrew/homebrew-core/blob/ab5040f62247ae337489653254bd817f33645006/Formula/swiftlint.rb#L24-L25)

Reply:
yeah sounds good to me to follow that and thank you 👍

Reply:
Done in 659b388c143fcebc88e679a424b78cbc9db03e15 👍